### ISSUE TITLE
"Local branches" when working with git trees

### ISSUE THREAD

**Issue by scallaway** [jesseduffield/lazygit#3561]
Discussion was the best label to put this under, but it's more a question for my understanding.

I've started using Worktrees as I finally have a decent use-case for them, but the one thing I'm noticing is that Lazygit seems to report all the branches that are on remote as "Local branches", even if they don't have any upstream associated with them.

In my case, this results in some 700 branches being listed in the "local branches", where I try and keep that as clean as possible (removing local branches that have been deleted on remote).

Is this purely because I cloned the repository with `--bare`? Or is it due to the interaction with worktrees. I didn't want to clone regularly and then work with worktrees as the additional folders it creates (for each worktree) would confuse things.

I've also tried running a `git remote prune origin` now that I've added an upstream, but that doesn't seem to sort things either.

Just some clarification on how this works would be great! (And anyway to clear out those excess branches)

**Comment by gfvioli** [comment]
Just today I ran into the same situation as you when testing bare repos on lazygit and personally I find it confusing. I think that I have a couple of suggestions to enhance the experience when working with worktrees on a bare repo:

- Ability to execute lazygit in the root of the bare repo, right now you have to go into one of the worktrees for lazygit to be able to launch. This is just a minor QoL not having to do cd once more and go to a worktree before calling lazygit. 

- I think in the case of a bare repo, worktrees currently located in panel 2 should replace local branches in panel 3. That way you avoid the confusion and all the clutter with local branches. Also allows for a speedier workflow since you don't have to be changing panel's 2 depth to see files status. 

To be honest, I'm not quite sure about the rationale to put worktrees on pane 2, my understanding is that while you can do worktrees on a NOT bare repo, as per ThePrimeagen video on it (link [here](https://www.youtube.com/watch?v=2uEqYw-N8uE)), the structure is a mess. Thus I think the recommendation is to use bare repos when using worktrees such that you can use them as independent branches, which would lead to this recommendation.

I'll keep on experimenting with bare repos and I'll keep on providing feedback.

**Comment by XavierChanth** [comment]
> Is this purely because I cloned the repository with `--bare`

Yes, all branches are "local" in a bare repo. `git clone --bare` is typically used for remote servers. My workaround:

`git clone --no-checkout -c core.bare=true`

This clones the repo to be setup in what I consider the best of both worlds:
- The git dir is ./.git/ instead of ./
- It doesn't pull all branches locally like a normal `git clone`
- It also doesn't create any work dirs like `git clone --bare`
- It properly configures the remote origin, since it's not treating itself as a remote repo (I think... I might still have something in my config which might do this automagically)
- It still otherwise looks and feels like a bare repo because it applies `core.bare=true` to the config in the end
